mongo/options/clientencryptionoptions.go

```diff
@@ -29,6 +36,43 @@ func (c *ClientEncryptionOptions) SetKmsProviders(providers map[string]map[strin
 	return c
 }
 
+// SetTLSConfig applies custom TLS options to configure connections with each KMS provider.
+func (c *ClientEncryptionOptions) SetTLSConfig(tlsOptsMap map[string]map[string]interface{}) (*ClientEncryptionOptions, error) {
```

Comment:
We may want to define an API that allows users to pass self-assembled `*tls.Config`s (like [ClientOptions.SetTLSConfig](https://github.com/mongodb/mongo-go-driver/blob/48ca24629ba874b6b64c3d460c446f71b72f9c2b/mongo/options/clientoptions.go#L720)) called `SetTLSConfig`. Rename `SetTLSConfig` to `SetTLSOptions` to align with the suggested API in the [KMS provider TLS options](https://github.com/mongodb/specifications/blob/ce242ddbb5377c509057e4e07b9d56c011bb0229/source/client-side-encryption/client-side-encryption.rst#kms-provider-tls-options) spec.

Also, a user may not understand what the expected values in `tlsOptsMap` are. Add a description of the expected format of `tlsOptsMap`, including the valid values for provider and for config values. Consider copying the comment from [ClientOptions.SetTLSConfig](https://github.com/mongodb/mongo-go-driver/blob/48ca24629ba874b6b64c3d460c446f71b72f9c2b/mongo/options/clientoptions.go#L697-L719).

Optional: Including the type in the variable name is redundant. Consider renaming `tlsOptsMap` to `tlsOpts`.

Reply:
Good suggestion, adding a more useful comment now.

Reply:
Nice, the description in `SetTLSOptions` is helpful.

The other proposal is to add another API `SetTLSOptions`:

```
func (c *ClientEncryptionOptions) SetTLSConfig (tlsConfig map[string]*tls.Config)
```

I was unsure of adding this originally. But I see two advantages and think it is worth considering:

- It helps with consistency of setting TLS options on a `mongo.Client`. TLS options can be set in the URI with `ClientOptions#ApplyURI` or with a tls.Config in `ClientOptions#SetTLSConfig`. Assuming there are use-cases for `ClientOptions#SetTLSConfig`, it seems reasonable to assume the same here.
- It enables an alternative for the workaround with testing PyKMIP. Instead of using the fork, tests can construct a `tls.Config` and specify a cipher that PyKMIP currently uses.

Reply:
Resolving. `SetTLSConfig` has been added